apps/dashboard/src/components/workflow-editor/steps/configure-step.tsx

```diff
@@ -11,13 +11,25 @@ import { motion } from 'framer-motion';
 import { InApp } from './in-app/in-app';
 import { useStep } from './use-step';
 import Chat from './chat';
+import { useState } from 'react';
+import { DeleteStepModal } from '@/components/workflow-editor/steps/delete-step-modal';
 
 export function ConfigureStep() {
+  const { step } = useStep();
+  const navigate = useNavigate();
   const { currentEnvironment } = useEnvironment();
-  const { workflowSlug = '' } = useParams<{
+  const { workflowSlug = '', stepSlug = '' } = useParams<{
     workflowSlug: string;
+    stepSlug: string;
   }>();
-  const { isReadOnly } = useWorkflowEditorContext();
+  const { isReadOnly, deleteStep } = useWorkflowEditorContext();
+
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+
+  const onDeleteStep = () => {
+    deleteStep(stepSlug);
+    navigate(buildRoute(ROUTES.EDIT_WORKFLOW, { environmentSlug: currentEnvironment?.slug ?? '', workflowSlug }));
```

Comment:
Focus back to workflow after opened step is deleted, otherwise the sidebar with deleted step will remain opened with undefined data.